modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main/categorization/categories/components/MoveCategoryTreeView.tsx

```diff
@@ -0,0 +1,498 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import ClayIcon from '@clayui/icon';
+import ClayModal from '@clayui/modal';
+import getCN from 'classnames';
+import {openToast} from 'frontend-js-components-web';
+import {fetch, sub} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
+
+import ApiHelper from '../../../../services/ApiHelper';
+import SpaceService from '../../../../services/SpaceService';
+import {FETCH_URLS} from './MoveCategoryModalContent';
+
+/**
+ * Returns info about the tree level.
+ * @param {number} depth Tree level in question
+ * @returns {object}
+ */
+const getTreeLevelInfo = (depth: number) =>
+	depth > 0
+		? {
+				getURL: FETCH_URLS.getSubCategories,
+				icon: 'categories',
+				showSelect: true,
+			}
+		: {
+				getURL: FETCH_URLS.getCategories,
+				icon: 'vocabulary',
+				showSelect: false,
+			};
+
+/**
+ * Returns a copy of the tree with properties added/updated to the object at index.
+ * @param {Array} tree An array of categories
+ * @param {number} index Position of object inside tree to update
+ * @param {object} properties Properties to add/edit within object at index
+ * @returns {Array}
+ */
+const getUpdatedTree = (tree: any, index: any, properties: any) => {
+	const treeCopy = tree.slice();
+
+	treeCopy.splice(index, 1, {...tree[index], ...properties});
+
+	return treeCopy;
+};
+
+const getSpaceName = async (assetLibrary: any) => {
+	if (assetLibrary?.some((item: any) => item === -1)) {
+		return 'All Spaces';
+	}
+	const spaces = await SpaceService.getSpaces().then((response) => {
+		return response.map((item) => ({
+			erc: item.externalReferenceCode,
+			label: item.name,
+			value: item.id,
+		}));
+	});
+
+	const sites = spaces.map((space: any) =>
+		ApiHelper.get<{items: any}>(
+			`/o/headless-site/v1.0/sites/by-external-reference-code/${space.erc}`
+		).then((response: {data: any}) => {
+			return response.data;
+		})
+	);
+
+	const fetchedSpaceData = await Promise.all(sites);
+
+	return assetLibrary?.map((item: any) => {
+		return fetchedSpaceData.find((space: any) => space.id === item).name;
+	});
```

Comment:
Does't the AssetLibrary already has the name info as defined in the [schema](https://github.com/liferay/liferay-portal/blob/master/modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/rest-openapi.yaml#L10-L14)? Why do we need to fetch all the spaces and sites to get the name?